crates/ads126x/src/register.rs

```diff
@@ -0,0 +1,94 @@
+use bitflags::bitflags;
+
+#[repr(u8)]
+pub enum Register {
+    ID = 0x00,
+    POWER = 0x01,
+    INTERFACE = 0x02,
+    MODE0 = 0x03,
+    MODE1 = 0x04,
+    MODE2 = 0x05,
+    INPMUX = 0x06,
+    OFCAL0 = 0x07,
+    OFCAL1 = 0x08,
+    OFCAL2 = 0x09,
+    FSCAL0 = 0x0A,
+    FSCAL1 = 0x0B,
+    FSCAL2 = 0x0C,
+    IDACMUX = 0x0D,
+    IDACMAG = 0x0E,
+    REFMUX = 0x0F,
+    TDACP = 0x10,
+    TDACN = 0x11,
+    GPIOCON = 0x12,
+    GPIODIR = 0x13,
+    GPIODAT = 0x14,
+    ADC2CFG = 0x15,
+    ADC2MUX = 0x16,
+    ADC2OFC0 = 0x17,
+    ADC2OFC1 = 0x18,
+    ADC2FSC0 = 0x19,
+    ADC2FSC1 = 0x1A,
+}
+
+bitflags! {
+    pub struct IdRegister: u8 {
+        const _ = !0; // Source may set any bits
+    }
+}
+
+pub enum DevId {
+    ADS1262 = 0b000,
+    ADS1263 = 0b001,
+}
+
+impl IdRegister {
+    pub fn get_rev_id(&self) -> u8 {
+        self.bits() & 0b0001_1111
+    }
+
+    pub fn get_dev_id(&self) -> DevId {
+        match (self.bits() & 0b1110_0000) >> 5 {
+            0b000 => DevId::ADS1262,
+            0b001 => DevId::ADS1263,
+
+            _ => panic!("Device ID must be 0b000 or 0b001."),
+        }
+    }
+}
+
+bitflags! {
+    pub struct PowerRegister: u8 {
+        const INTREF = 0b0000_0001;
+        const VBIAS  = 0b0000_0010;
+        const RESET  = 0b0001_0000;
+    }
+}
+
+bitflags! {
+    pub struct InterfaceRegister: u8 {
+        const STATUS  = 0b0000_0100;
+        const TIMEOUT = 0b0000_1000;
+
+        const _ = 0b0000_0011; // Source may set CDC bits
+    }
+}
+
+pub enum Crc {
+    DISABLED = 0b00,
+    ENABLED = 0b01,
+    RESERVED = 0b11,
```

Comment:
`RESERVED` is reserved by the ADS126x. Should this instead panic rather than being part of the enum?

Reply:
Hmm ideally the code just doesn't compile if you try to use it. I'd say that it shouldn't exist as an option, unless for some reason it could exist for a valid reason, but I don't think it will.

Reply:
So theoretically I think the ADC can set the bits to reserved, but I don't really see what that does. Can you let me know what your interpretation is based on the Register section of the SDS please? If you think it's fine to error, I'll remove the reserved and make it panic. Otherwise, if the ADC actually _might_ set the crc to reserved, we don't want to crash.

Reply:
I think it depends, I'd liken the RESERVED register to a zero register on ARM, but we don't need to interact with that. The better example for this situation I have is MODE0 filter, where you see reserved, The idea I think is that they've just reserved these as zero since they don't have enough options to fill all of them and nothing else to put into that register so they've "reserved" them. I think we should just take the option away for the user to interact with these bits.

Reply:
Also, for reserved in the context of bit 7 of MODE1 register where it is reserved an only should ever be 0 then we take away the ability of the user to set this bit to something other than zero.